pkg/clusteragent/admission/controllers/webhook/common.go

```diff
@@ -28,12 +30,32 @@ func buildLabelSelector() *metav1.LabelSelector {
 				},
 			},
 		}
+	} else {
+		// Ignore all, accept pods if they're explicitly allowed
+		labelSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				common.EnabledLabelKey: "true",
+			},
+		}
 	}
 
-	// Ignore all, accept pods if they're explicitly allowed
-	return &metav1.LabelSelector{
-		MatchLabels: map[string]string{
-			common.EnabledLabelKey: "true",
-		},
+	labelSelector.MatchExpressions = append(
+		labelSelector.MatchExpressions,
+		azureAKSLabelSelectorRequirement(),
+	)
```

Comment:
Based on [azure's docs](https://docs.microsoft.com/en-us/azure/aks/faq#can-i-use-admission-controller-webhooks-on-aks), this should only be added as a namespace selector, while this function `buildLabelSelector` return value can be used as object selector (default), so we're adding the rule unnecessarily for object selectors.
I think we should adapt the code to only add this aks rule to the namespace selector.

Reply:
Yes. You're right :+1: